americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+            p_values = np.hstack((p_values,
+                                  _correct_p_value(ref_less, p_value,
+                                                   dist_bar[ref_loc],
+                                                   dist_bar[id2])))
+
+        dist_bar = np.array(dist_bar)
+        dist_std = np.array(dist_std)
+
+        # Creates the boxplot
+        ax, xpos = barchart(dist_bar,
+                            ax=ax,
+                            errors=dist_std,
+                            interval=interval,
+                            width=width,
+                            offset=offset,
+                            xticklabels=order,
+                            colormap=colormap,
+                            match_colors=match_colors,
+                            elinewidth=elinewidth,
+                            ecapwidth=ecapwidth)
+        # Watches the xlimits
+        if id1 == 0:
+            xlim = [xpos[0] - interval*0.75]
+
+        # Gets the critical lines to display on the figure
+        if lowest is None:
+            lowest, __ = _get_bar_height(dist_bar+dist_std)
+        if p_values is not None and (ref_loc == 0 or len(dist_bar) == 2):
+            bars = add_comparison_bars(xpos,
+                                       dist_bar+dist_std,
+                                       p_values,
+                                       ax,
+                                       lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+        elif p_values is not None and ref_loc == (len(dist_bar) - 1):
+            bars = add_comparison_bars(xpos[::-1],
+                                       (dist_bar+dist_std)[::-1],
+                                       p_values[::-1],
+                                       ax,
+                                       lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+        elif p_values is not None:
+            bars = add_comparison_bars(xpos[ref_loc:],
+                                       dist_bar[ref_loc:] + dist_std[ref_loc:],
+                                       p_values[(ref_loc):],
+                                       ax, lowest=lowest,
+                                       label_size=label_size,
+                                       show_value=show_p_value)
+            bars2 = add_comparison_bars(xpos[:(ref_loc+1)][::-1],
+                                        dist_bar[:(ref_loc+1)][::-1] +
+                                        dist_std[:(ref_loc+1)][::-1],
+                                        p_values[:(ref_loc)][::-1],
+                                        ax, lowest=lowest,
+                                        label_size=label_size,
+                                        show_value=show_p_value)
+            bars.extend(bars2)
+
```

Comment:
If a `p_table` wasn't passed, `p_values` can be `None`.

In theory it's there for a quick, dirty approach to look at the data without spending time on make_distance_boxplots.